sql-plugin/src/main/java/com/nvidia/spark/rapids/iceberg/spark/source/GpuIcebergReader.java

```diff
@@ -158,7 +159,7 @@ static ColumnarBatch addUpcastsIfNeeded(ColumnarBatch batch, Schema expectedSche
         GpuColumnVector oldColumn = columns[i];
         columns[i] = GpuColumnVector.from(
             GpuCast.doCast(oldColumn.getBase(), oldColumn.dataType(), expectedSparkType,
-            false, false, false), expectedSparkType);
+            new SparkConfCastOptions(false, false, false)), expectedSparkType);
```

Comment:
Calling a val from a scala object wasn't very obvious to me, I have updated the code to use the `DEFAULT_CAST_OPTIONS` It's hard to read but it will not create a new object everytime